src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs

```diff
@@ -206,7 +206,10 @@ private void WriteMethodBody(IMethodDefinition method)
                 Write("throw new ");
                 if (_forCompilationIncludeGlobalprefix)
                     Write("global::");
-                Write("System.PlatformNotSupportedException(); ");
+                if(_platformNotSupportedExceptionMessage == null)
+                    Write("System.PlatformNotSupportedException();");
+                else
+                    Write($"System.PlatformNotSupportedException(\"{_platformNotSupportedExceptionMessage}\"); ");
```

Comment:
@ericstj Wouldn't we also need to build resource generation into this task?  I assume it currently doesn't bother including SR at all (but I could easily be wrong)